File: Answer/2126.摧毁小行星.java
/*
 * @lc app=leetcode.cn id=2126 lang=java
 *
 * [2126] 摧毁小行星
 */

// @lc code=start
import java.util.Arrays;

class Solution {

    public boolean asteroidsDestroyed(int mass, int[] asteroids) {
        Arrays.sort(asteroids);
        long currentMass = mass;
        for (int asteroid : asteroids) {
            if (currentMass < asteroid) {
                return false;
            }
            currentMass += asteroid;
        }
        return true;
    }
}
// @lc code=end

File: Answer/2126.摧毁小行星.js
/*
 * @lc app=leetcode.cn id=2126 lang=javascript
 *
 * [2126] 摧毁小行星
 */

// @lc code=start
/**
 * @param {number} mass
 * @param {number[]} asteroids
 * @return {boolean}
 */
var asteroidsDestroyed = function (mass, asteroids) {
  asteroids.sort((a, b) => a - b);
  let currentMass = BigInt(mass);
  for (const asteroid of asteroids) {
    const asteroidMass = BigInt(asteroid);
    if (currentMass < asteroidMass) {
      return false;
    }
    currentMass += asteroidMass;
  }
  return true;
};
// @lc code=end
